metagpt/roles/role.py

```diff
@@ -524,6 +524,7 @@ def is_idle(self) -> bool:
 
     async def think(self) -> Action:
         """The exported `think` function"""
+        await self._observe()
```

Comment:
why _observe here?

Reply:
pending test

Reply:
There are still some counter-instincts in the think _observe, can you add the reason to the comments?